ts/Series/ArcDiagram/ArcDiagramSeries.ts

```diff
@@ -0,0 +1,603 @@
+/* *
+ *
+ *  Arc diagram module
+ *
+ *  (c) 2021 Piotr Madej, Grzegorz Blachliński
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type ArcDiagramSeriesOptions from './ArcDiagramSeriesOptions';
+import ArcDiagramPoint from './ArcDiagramPoint.js';
+
+import SankeyColumnComposition from '../Sankey/SankeyColumnComposition.js';
+import SeriesRegistry from '../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        sankey: SankeySeries
+    }
+} = SeriesRegistry;
+import U from '../../Core/Utilities.js';
+const {
+    extend,
+    merge,
+    pick,
+    relativeLength
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.arcdiagram
+ *
+ * @augments Highcharts.seriesTypes.sankey
+ */
+class ArcDiagramSeries extends SankeySeries {
+
+    /**
+     *  Arc diagram series is a chart drawing style in which
+     *  the vertices of the chart are positioned along a line
+     *  on the Euclidean plane and the edges are drawn as a semicircle
+     *  in one of the two half-planes delimited by the line,
+     *  or as smooth curves formed by sequences of semicircles.
+     *
+     * @sample highcharts/demo/arc-diagram/
+     *         Arc Diagram
+     *
+     * @extends      plotOptions.sankey
+     * @exclude      dataSorting
+     * @since        next
+     * @product      highcharts
+     * @requires     modules/arc-diagram
+     * @optionparent plotOptions.arcdiagram
+     */
+    public static defaultOptions: ArcDiagramSeriesOptions = merge(SankeySeries.defaultOptions, {
+        nodeShape: 'circle',
```

Comment:
Missing doclet.

Would it make sense to reuse `marker` API for nodes? Then use the marker's logic which is already implemented (color, stroke etc.)

Reply:
Partially done the migration to the `marker` (`symbol` might be used now) but the rest would require a change of the concept of how nodes are drawn - currently, it is an extension of the `sankey` which is an extension of the `column` series.

Reply:
Aaah, right, thanks 👍